### ISSUE TITLE
Resizing window of example app breaks layout and generates TypeError

### ISSUE THREAD

**Issue by HalfWhitt** [beeware/toga-chart#191]
### Describe the bug

Dragging the edges of  the window to resize the example app (at least on macOS) generates repeated instances of this error:

```
Exception ignored on calling ctypes callback function <rubicon.objc.api.objc_method object at 0x10718c600>:
Traceback (most recent call last):
  File "/Users/charles/.pyenv/versions/toga_313/lib/python3.13/site-packages/rubicon/objc/api.py", line 339, in __call__
    result = self.py_method(py_self, *args)
  File "/Users/charles/toga_dev/toga/cocoa/src/toga_cocoa/widgets/canvas.py", line 41, in drawRect_
    self.interface.context._draw(self.impl, draw_context=context)
  File "/Users/charles/toga_dev/toga/core/src/toga/widgets/canvas.py", line 461, in _draw
    obj._draw(impl, **kwargs)
  File "/Users/charles/toga_dev/toga/core/src/toga/widgets/canvas.py", line 1070, in _draw
    obj._draw(impl, **sub_kwargs)
  File "/Users/charles/toga_dev/toga/core/src/toga/widgets/canvas.py", line 383, in _draw
    impl.write_text(
  File "/Users/charles/toga_dev/toga/cocoa/src/toga_cocoa/widgets/canvas.py", line 305, in write_text
    rs = self._render_string(line, font, **kwargs)
  File "/Users/charles/toga_dev/toga/cocoa/src/toga_cocoa/widgets/canvas.py", line 247, in _render_string
    textAttributes[NSFontAttributeName] = font.native
TypeError: 'ObjCInstance' object does not support item assignment
```
This happens when resizing either horizontally, vertically, or both. Additionally, reducing vertical size (not necessarily smaller than initial — any reduction from current) creates weird rendering overlaps.

Notably, this *doesn't* happen when first rendering the window and widget. And it doesn't *always* happen when resizing, either, only intermittently.

### Steps to reproduce

1. Go to the examples folder and run `python -m chart`
2. Resize the window.

### Expected behavior

No exception, and proper layout/rendering

### Screenshots

![image](https://github.com/user-attachments/assets/354ddc62-75ab-4694-9d14-b31f79f3f988)

### Environment

- Operating System: macOS
- Python version: 3.13
- Software versions:
  - Toga: current main branch
  - Travertino: both 0.3.0 and current main branch

**Comment by freakboy3742** [comment]
This is a weird one. It looks like it's deep in Rubicon.

The code in question is doing the following:

```
textAttributes = NSMutableDictionary.alloc().init()
textAttributes[NSFontAttributeName] = font.native
```

On a normal (successful) invocation, `type(textAttributes)` returns:
```
type[<ObjCMutableDictInstance: __NSDictionaryM at 0x130ada330: {}]
```
However, on the problem call, the same call returns:
```
type[<ObjCInstance: __NSDictionaryM at 0x12a2341a0: {}>]
```

So - it's not identifying that the MutableDict wrapper needs to be applied to the object returned by the constructor.

There's an immediate fix in converting all the use of subscript notation into `setObject:forKey:` calls; message passing works on any object. But that's not a very satisfying answer, as the underlying Rubicon issue still exists.  

I wonder if this might be a niche manifestation of beeware/rubicon-objc#256 - if the ObjC memory address is being recycled before Python disposes of the object, that would result in the wrong object wrapper being applied, resulting in the `__setitem__` handler not being available. It's only manifesting here because there are *lots* of calls to render strings, which means it's more likely to expose this sort of memory recycling issue.

**Comment by freakboy3742** [comment]
Following up on this - confirming that it *is* a Rubicon issue, and one that will be resolved by beeware/rubicon-objc#543.
